driver/src/main/java/org/kaazing/k3po/driver/internal/behavior/handler/event/AbstractEventHandler.java

```diff
@@ -99,14 +99,7 @@ protected void handleUpstream1(ChannelHandlerContext ctx, ChannelEvent evt) thro
             handleUnexpectedEvent(ctx, evt);
         } else {
             ChannelFuture pipelineFuture = getPipelineFuture();
-            if (!pipelineFuture.isDone()) {
-                pipelineFuture.addListener(new ChannelFutureListener() {
-                    @Override
-                    public void operationComplete(ChannelFuture future) throws Exception {
-                        handleUpstream1(ctx, evt);
-                    }
-                });
-            } else if (!pipelineFuture.isSuccess()) {
+            if (!pipelineFuture.isSuccess()) {
```

Comment:
Is this correct ? We don't know pipelineFuture is completed or not at this point. So pipelineFuture.isSuccess() is false doesn't mean a failure, right ?